### ISSUE TITLE
Allow for the creation of multiple environments with custom names

### ISSUE THREAD

**Issue by goanpeca** [conda/constructor#359]
Would it be possible to update constructor to do something like this:

```yaml
name: Uberconda
version: 1.0
install_in_dependency_order: True
channels:
  - http://repo.anaconda.com/pkgs/main/

specs:
    - python 3.5*
    - conda

envs:
  - name: something_else
    specs:
    - python 3.5*
    - numpy
  - name: _something_private
    specs:
    - python 3.5*
    - pandas

license_file: EULA.txt

# Welcome image for Windows installer
welcome_image: bird.png                  [win]
```

**Comment by mcg1969** [comment]
Working on it! #356

**Comment by goanpeca** [comment]
Hi @mcg1969, thanks for the feedback. 

It seems that PR is allowing to create multiple installers with a single environment on once go. Did I get this right?

What I am asking for is the possibility of having an installer that will create several environments, not just a `base` one.

Thanks!

**Comment by mcg1969** [comment]
Oh I see, that's different! Sorry for the confusion

**Comment by mcg1969** [comment]
Thinking about this more, this would definitely be _possible_ but quite challenging. Here's what it would need to do:

- Run the conda solver on each environment and collect the package lists for each.
- Merge and deduplicate into into a single package list. Use this list to build the package cache blob.
- Use single `--extract-tarball` and `--extract-conda-pkgs` to unpack the resulting package cache 
- Create multiple `conda install` commands, one for each environment.

Open questions:
- One `pre_install` per environment, or one per installer?
- One `post_install` per environment, or one per installer?
- How to differentiate between root and child?

**Comment by goanpeca** [comment]
Hi @mcg1969 

>this would definitely be possible but quite challenging.

🤷  🙃 

>These would need to be done for all installer types. I'm not inclined for us to build something here that is unique to Unix, say.

Agreed. 

The idea of a 1 click installer is something I want to make happen (sooner than later) for projects like JupyterLab. I want to have the actual applications be installed on "private" envs `_jupyterlab` and provide a safe update path and way for installing extensions and additional kernels, and also connecting to any other envs in the system. 

This idea initially requires having at least 2 envs, a `base` one and the application one `_jupyterlab` 

>One pre_install per environment, or one per installer?

I would say per installer since am installer acts as a whole, just that it is using more than 1 environment to handle different things.

>One post_install per environment, or one per installer?

Same as before

>How to differentiate between root and child?

In terms of what?

Thanks!

**Comment by mcg1969** [comment]
At least one of the environments in the list needs to be the root of the installed environment tree. So how do we specify that in a case like this?￼

**Comment by goanpeca** [comment]
In the example I gave above https://github.com/conda/constructor/issues/359#issue-683908391

I added a key called, default, it could be base or root or similar, and there could only be one set to `True`.

**Comment by mcg1969** [comment]
That works!

**Comment by goanpeca** [comment]
Thanks @mcg1969 :-), please let me know in any way I can help. Also my suggestion was really a quick idea, not thoroughly designed, so take with a grain of 🧂

**Comment by mcg1969** [comment]
Well to be clear, this is not something I personally am going to be able to implement—so if you or anyone else wants to take a crack at it, please do! I am happy to offer reviews and guidance.

**Comment by goanpeca** [comment]
I would like to hear more use cases, I have a very specific one in mind :-) 

Any guidance is also welcome so I could also tackle this at some point 🤷

**Comment by goanpeca** [comment]
@jaimergp

**Comment by jaimergp** [comment]
Hey folks, this feature has been implemented in #509 if you want to take a look!
